### DISCUSSION TITLE
branch-heterogeneity in cogent3 model instances

### DISCUSSION THREAD

**Discussion by KatherineCaley** [cogent3/cogent3#933] [Q&A]
How do you construct a cogent3 model app instance which allows rate parameters to be different between edges?

**Comment by GavinHuttley** [comment]
Thanks for asking!

This example assumes you only have 3 sequences (so the tree can be correctly assumed as star-shaped). Please note that the `brca1.fasta` alignment is available under `tests/data/brca1.fasta`.

```python
from cogent3 import load_aligned_seqs
from cogent3.app import evo

aln = load_aligned_seqs("~/repos/Cogent3/tests/data/brca1.fasta", moltype="dna")

# making sure we only have 3 sequences
aln = aln.take_seqs(["Human", "Rhesus", "Dog"])
```
We now define the model app using the HKY85 model (but any `cogent3` parametric model that has rate parameters will work). I'm also going set an upper bound for the rate parameters (experience shows this can guide the optimiser away from silly local maxima).
```python
mod_dependent = evo.model("HKY85", time_het=[dict(edges=["Human", "Rhesus"], is_independent=False, upper=100)])
```
Using it is just
```python
result = mod_dependent(aln)
print(result.lf)
```
Which generates
```
HKY85
log-likelihood = -6090.1694
number of free parameters = 5
====================================
edge      parent    length     kappa
------------------------------------
Human     root      0.0179    5.6948
Rhesus    root      0.0247    5.6948
Dog       root      0.1882    4.2709
------------------------------------
====================================
     A         C         G         T
------------------------------------
0.3686    0.1782    0.2150    0.2381
------------------------------------
```
You can see the rate parameter `kappa` is the same for Human / Rhesus and different to that of Dog.

**Reply by KatherineCaley** [reply]
Perfect! Thank you!
